services/blockchain-app-registry/shared/metadata.js

```diff
@@ -77,24 +77,32 @@ const getBlockchainAppsMetaList = async (params) => {
 		}];
 	}
 
-	const limit = params.limit * config.supportedNetworks.length;
+	let { offset } = params;
+	let { limit } = params;
 	const defaultApps = await applicationMetadataTable.find(
-		{ ...params, limit, isDefault: true },
+		{ ...params, isDefault: true },
 		['network', 'chainID', 'chainName'],
 	);
 
-	if (defaultApps.length < params.limit) {
+	if (defaultApps.length < limit) {
+		// Update offset and limit
+		limit -= defaultApps.length;
+		offset -= await applicationMetadataTable.count(
+			{ ...params, isDefault: true },
+		);
+
+		if (offset < 0) offset = 0;
```

Comment:
Would it be more appropiate to substract total number of default apps from the offset? suggesting just an alternative way of writing.

```suggestion
		const totalDefaultAppsCount = await applicationMetadataTable.count(
			{ isDefault: true },
		);

		offset = Math.max(totalDefaultAppsCount, 0);
```